crates/biome_css_analyze/src/lint/nursery/no_invalid_direction_in_linear_gradient.rs

```diff
@@ -50,39 +50,20 @@ declare_rule! {
 
 lazy_static! {
     // It is necessary to find case-insensitive string.
-    // For example, both 'linear-gradinet' and 'Linear-gradient' should pass the check.
-    pub static ref LINEAR_GRADIENT_FUNCTION_NAME: Regex =
-        Regex::new(r"^(?i)(-webkit-|-moz-|-o-|-ms-)?linear-gradient").unwrap();
-
-    // It is necessary to find case-insensitive string.
-    // Also Check if 'in' is a word.
+    // Also Check if 'in' is a word boundary.
     // For examples,`to top in srgb` is valid but `to top insrgb` is not valid.
     pub static ref IN_KEYWORD: Regex = Regex::new(r"(?i)\bin\b").unwrap();
 
     // This regex checks if a string consists of a number immediately followed by a unit, with no space between them.
+    // For examples, `45deg`, `45grad` is valid but `45 deg`, `45de` is not valid.
     pub static ref ANGLE: Regex = Regex::new(r"^[\d.]+(?:deg|grad|rad|turn)$").unwrap();
 
-    // It is necessary to find case-insensitive string.
-    // For example, both 'top' and 'TOP' should pass the check.
-    pub static ref DIRECTION: Regex = Regex::new(r"(?i)top|left|bottom|right").unwrap();
-
-    // This need for capture 'side-or-corner' keyword from linear-gradient function.
-    // Ensure starts with the keyword 'to' and ends with the keyword 'side-or-corner'.
-    pub static ref DIRECTION_WITH_TO: Regex = Regex::new(&format!(
-        r"(?i)^to ({})(?: ({}))?$",
-        DIRECTION.as_str(),
-        DIRECTION.as_str()
-    ))
-    .unwrap();
-
-    // This need for capture 'side-or-corner' keyword from linear-gradient function
-    // Ensure starts with the keyword 'side-or-corner' and ends with the keyword 'side-or-corner'.
+    // This need for capture `side-or-corner` keyword from linear-gradient function.
+    // Ensure starts `side-or-corner` keyword `to` and ends with the keyword `side-or-corner`.
     pub static ref DIRECTION_WITHOUT_TO: Regex = Regex::new(&format!(
```

Comment:
This regular expression is written in the following syntax and is used to capture strings located at `side-or-corner`.
[mdn side-or-corner](https://developer.mozilla.org/en-US/docs/Web/CSS/gradient/linear-gradient)
```
<linear-gradient-syntax> =
[ <angle> | to <side-or-corner> ]? , <color-stop-list>  
<side-or-corner> =
[ left | right ] ||
[ top | bottom ]
```
**examples
valid**

- top left
- top right

**invalid**

- to left
- top righ
- topright